src/_P039_Thermosensors.ino

```diff
@@ -1938,4 +1938,78 @@ void transfer_n_ByteSPI(uint8_t l_CS_pin_no, uint8_t l_noBytesToSend, uint8_t* l
 
 }
 
+/**************************************************************************/
+/*!
+    @brief read a 16Bit register and change a flag, writing it back, handling a GPIO CS
+    @param l_CS_pin_no the GPIO pin number used as CS
+    @param l_readaddress SPI read address of the device register
+    @param l_writeaddress SPI write address of the device register
+    @param l_flagmask mask set to apply on the read register
+    @param l_set_reset controls if flag mask will be set (-> true) or reset ( -> false)
+    
+
+    @returns
+
+    Initial Revision - chri.kai.in 2021 
+
+/**************************************************************************/
+
+void change16BitRegister(uint8_t l_CS_pin_no, uint8_t l_readaddress, uint8_t l_writeaddress, uint16_t l_flagmask, bool l_set_reset )
+{
+  uint16_t l_reg = 0u;
+
+  // read in config register
+  l_reg = read16BitRegister(l_CS_pin_no, l_readaddress);
+
+  if(l_set_reset){
+    l_reg |= l_flagmask;
+  }
+  else
+  {
+    l_reg &= ~(l_flagmask);
+  }
+  
+  // write to configuration register
+  write16BitRegister(l_CS_pin_no, l_writeaddress, l_reg);
+
+
+}
+
+/**************************************************************************/
+/*!
+    @brief read a 8 Bit register and change a flag, writing it back, handling a GPIO CS
+    @param l_CS_pin_no the GPIO pin number used as CS
+    @param l_readaddress SPI read address of the device register
+    @param l_writeaddress SPI write address of the device register
+    @param l_flagmask mask set to apply on the read register
+    @param l_set_reset controls if flag mask will be set (-> true) or reset ( -> false)
+    
+
+    @returns
+
+    Initial Revision - chri.kai.in 2021 
+
+/**************************************************************************/
+
+void change8BitRegister(uint8_t l_CS_pin_no, uint8_t l_readaddress, uint8_t l_writeaddress, uint8_t l_flagmask, bool l_set_reset )
+{
+  uint8_t l_reg = 0u;
+
+  // read in config register
+  l_reg = read8BitRegister(l_CS_pin_no, l_readaddress);
+
+  if(l_set_reset){
+    l_reg |= l_flagmask;
+  }
+  else
+  {
+    l_reg &= ~(l_flagmask);
+  }
```

Comment:
Using a mask can be done, but you can use the Arduino macros to make life easier:
```c++
#define bitRead(value, bit) (((value) >> (bit)) & 0x01)
#define bitSet(value, bit) ((value) |= (1UL << (bit)))
#define bitClear(value, bit) ((value) &= ~(1UL << (bit)))
#define bitWrite(value, bit, bitvalue) (bitvalue ? bitSet(value, bit) : bitClear(value, bit))
```
As you can see, those are for max. 32-bit sized unsigned integers, but at least they make the code a lot easier to read.

Reply:
I sense some opportunity here. For the moment i tend to stick to the mechanism until i understood the side effects to the compelte code. On my first glimpse, using the macros would require a slightly different definition of the masks as i did it so far. (no. of adressed bit, instead of dedicated value). I will add this as a feature request to my list.

Reply:
Yep, it indeed requires changing the mask defines, which can result in new bugs.
Having it in a separate commit is a good idea.

Reply:
Having an active shifting with every application of a mask is no problem for the CPU Load ? Sounds like "CPU power galore" to me, but yes, ESP  is no 8-bit controller - i tend to forget.

Reply:
The compiler can do nifty tricks here, as it isn't a function, but a define.
So if the `bit` value is a `constexpr`, the shift is already computed at compile time.
Also bit shifting is 1 instruction and you already have to load the data into the registers anyway.

Apart from that the functions are not called 100's times per second.
It is good to keep optimizations in mind, just don't start with it ;)

Reply:
I mark this finding as TODO: in the code to not forget and leave it for the moment as it is.